pkg/client/kubernetes/apply.go

```diff
@@ -121,32 +121,82 @@ func (c *Applier) deleteObject(ctx context.Context, desired *unstructured.Unstru
 var DefaultApplierOptions = ApplierOptions{
 	MergeFuncs: map[schema.GroupKind]MergeFunc{
 		corev1.SchemeGroupVersion.WithKind("Service").GroupKind(): func(newObj, oldObj *unstructured.Unstructured) {
-			// We do not want to overwrite a Service's `.spec.clusterIP', `.spec.healthCheckNodePort` or '.spec.ports[*].nodePort' values.
-			oldPorts := oldObj.Object["spec"].(map[string]interface{})["ports"].([]interface{})
-			newPorts := newObj.Object["spec"].(map[string]interface{})["ports"].([]interface{})
-			ports := []map[string]interface{}{}
-
-			// Check whether ports of the newObj have also been present previously. If yes, take the nodePort
-			// of the existing object.
-			for _, newPort := range newPorts {
-				np := newPort.(map[string]interface{})
-
-				for _, oldPort := range oldPorts {
-					op := oldPort.(map[string]interface{})
-					// np["port"] is of type float64 (due to Helm Tiller rendering) while op["port"] is of type int64.
-					// Equality can only be checked via their string representations.
-					if fmt.Sprintf("%v", np["port"]) == fmt.Sprintf("%v", op["port"]) {
-						if nodePort, ok := op["nodePort"]; ok {
-							np["nodePort"] = nodePort
+			newSvcType, found, _ := unstructured.NestedString(newObj.Object, "spec", "type")
+			if !found {
+				newSvcType = string(corev1.ServiceTypeClusterIP)
+				_ = unstructured.SetNestedField(newObj.Object, newSvcType, "spec", "type")
+			}
+
+			oldSvcType, found, _ := unstructured.NestedString(oldObj.Object, "spec", "type")
+			if !found {
+				oldSvcType = string(corev1.ServiceTypeClusterIP)
+
+			}
+
+			switch newSvcType {
```

Comment:
Imo, no `switch` necessary because the second `case` in line 172 is obsolet if you use an `if` condition for the first case.

Reply:
Then I'll have to do 

```go
if newSvcType== string(corev1.ServiceTypeLoadBalancer) && 
  newSvcType==string(corev1.ServiceTypeNodePort) &&
  newSvcType!=string(corev1.ServiceTypeClusterIP) { ..
```

switch makes more sense in this case